### PR TITLE
document: display cover thumbnail

### PR DESCRIPTION
**Pull request by AoNoOokami**
* Moves thumbnail logic in 'shared' library as it's used in both admin
and public-search projects.
* Adds thumbnails in professional brief and detailed view.
* Moves 'type' field below the thumbnail in admin detailed view.
* Closes rero/rero-ils#1188.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>

## Why are you opening this PR?

To display thumbnails in brief and detail views of both admin and public parts of the application.
This is a part of US1812.

## Dependencies

My PR depends on `rero-ils#<xx>`'s PR(s):

N/A

## How to test?

Checkout this PR locally.
`npm i` to install shared library
`npm run start-admin-proxy` to check admin part
`npm run start-public-search-proxy` to check public part

Check thumbnail display in both projects and both views (brief and detailed) on localhost:4200.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 1]
```diff
@@ -26,122 +26,124 @@
...
+      <!-- TITLE -->
+      <h2 id="doc-title" *ngIf="record.metadata.title && record.metadata.title | mainTitle as title" class="pl-0">
+        <ng-core-text-read-more
+          [text]="title"
+          [unit]="'character'"
+          [limit]="150"
+          [showMoreLabel]="'Show more' | translate"
+          [showLessLabel]="'Show less' | translate"
+        ></ng-core-text-read-more>
+      </h2>
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:44]: indentation problem. For very large file, a good indentation is a "must-have"

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 2]
```diff
@@ -26,122 +26,124 @@
...
+
+      <!-- EDITION STATEMENT -->
+      <ng-container *ngIf="record.metadata.editionStatement && record.metadata.editionStatement.length > 0">
         <ul class="list-unstyled mb-0">
-          <li *ngFor="let value of provision.value; let i = index"
-          [attr.id]="i | idAttribute:{prefix: 'doc-publication-statement'}">
-            {{ value }}
+          <li *ngFor="let edition of getStatement(record.metadata.editionStatement); let i = index"
+          [attr.id]="i | idAttribute:{prefix: 'doc-edition-statement'}">
+          {{ edition.value }}
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:94]: here too (this seems cause a lot of problem to `git diff`)

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  ngOnInit() {
+    this.coverUrl = `/static/images/icon_${this.record.metadata.type}.png`;
+    if (this.record.metadata.cover_art) {
+      this.coverUrl = this.record.metadata.cover_art;
+    } else if (this.record.metadata.identifiedBy) {
+      for (const identifier of this.record.metadata.identifiedBy) {
+        if (identifier.type === 'bf:Isbn') {
+          this.isbn = identifier.value;
+        }
+      }
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:97]:
```typescript
this.isbn = this.record.metadata.identifyBy
    .filter(id => id.type === 'bf:Isbn')  // keep only isbn
    .map(id => id.value)  // get identifier value
    .find()  // return the first array value or undefined if array is empty
```
To be tested in real situation. For my point of view, it's more javascript minded (and if I had a good memory, it's more efficient for CPU)

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      if (this.isbn) {
+        this.getCoverUrl();
+      }
+    }
+    console.log(this.isSmall);
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:102]: can be removed

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    if (this.isSmall) {
+      this.figureStyle = 'thumb-brief';
+
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:105]: empty line

**Review by zannkukai — CHANGES_REQUESTED** [review]
Some comment are cosmetics ;-)

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+<div class="align-self-start d-flex justify-content-center">
+  <figure class="mb-0" [ngClass]="figureStyle">
+    <ng-container *ngIf="record.metadata.cover_art; else noLink">
+      <a href="coverUrl"><img [ngClass]="imageCssClass" [src]="coverUrl"></a>
+    </ng-container>
+    <ng-template #noLink>
+      <img [ngClass]="imageCssClass" [src]="coverUrl">
+    </ng-template>
+    <figcaption class="text-center">{{ record.metadata.type | translate }}</figcaption>
```
Comment (Garfield-fr) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.html:26]: `<figcaption class="text-center" transate>{{ record.metadata.type }}</figcaption>`

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<!--
+ RERO ILS UI
+ Copyright (C) 2019 RERO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.html:3]: `2020`

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        }
+      })
+    );
+  }
+
+  /**
+   * Get cover url
+   * @returns string - url of the cover if cover exists.
+   */
+  getCoverUrl() {
```
Comment (Garfield-fr) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:76]: Group the 2 functions into 1: `getCoverImage` and `getCoverUrl` is overkill

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 9]
```diff
@@ -26,122 +26,124 @@
...
   <!-- HEADER -->
-  <header class="col-10 mt-3 mb-3">
-    <!-- TYPE -->
-    <ng-container *ngIf="record.metadata.type">
-      <span id="doc-type">{{ record.metadata.type | translate | titlecase }}</span>
-    </ng-container>
+  <header class="col-10">
+    <div class="row mt-3 mb-3">
+      <div class="col-2">
+        <shared-thumbnail [record]="record"></shared-thumbnail>
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:32]: Added the class parameter to be more flexible. The CSS definition stays in the main application.

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 10]
```diff
@@ -26,122 +26,124 @@
...
-        [unit]="'character'"
-        [limit]="150"
-        [showMoreLabel]="'Show more' | translate"
-        [showLessLabel]="'Show less' | translate"
-      ></ng-core-text-read-more>
-    </h2>
+      <!-- TITLE ALTERNATIVE GRAPHICS -->
+      <ng-container *ngIf="record.metadata.ui_title_altgr">
+        <ng-container *ngFor="let altgr_title of record.metadata.ui_title_altgr; let i = index">
+          <h3 id="{{'doc-altgr-title-' + i}}">{{ altgr_title }}</h3>
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:49]: Missing space before ` }}` and after `{{`

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 11]
```diff
@@ -26,122 +26,124 @@
...
-                </ul>
-              </ng-container>
-            </div>
-          </dd>
-        </dl>
+      <!-- IS PART OF -->
+      <ng-container *ngFor="let document of record.metadata.partOf; let i = index">
+        <ng-container *ngIf="document.document.pid | getRecord: 'documents' : 'object' : '': { 'Content-Type': 'application/rero+json' } | async as hostDocument">
+          <dl class="row mb-0">
+            <dt id="{{'doc-part-of-label-' + i}}" class="col-auto">
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:103]: Missing space before ` }}` and after `{{`

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 12]
```diff
@@ -26,122 +26,124 @@
...
+            <dd class="col mb-0">
+              <div class="row">
+                <a id="{{'doc-part-of-' + i}}"[routerLink]="['/records', 'documents', 'detail', document.document.pid]">
+                  {{ getShortMainTitle(hostDocument.metadata.title) }}
+                </a>
+                <ng-container *ngIf="document.numbering">
+                  <span>;</span>
+                  <ul class="list-unstyled mb-0 ml-1">
+                    <li *ngFor="let numbering of document.numbering">
+                      <span id="{{'doc-part-of-numbering-' + i}}" *ngIf="formatNumbering(numbering) as num">{{ num }}</span>
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:115]: Missing space before ` }}` and after `{{`

File: projects/public-search/src/app/document-brief/document-brief.component.html [hunk 13]
```diff
@@ -16,62 +16,55 @@
...
+        <span translate *ngIf="record.metadata.available">available</span>
+        <span translate *ngIf="!record.metadata.available">not available</span>
+      </div>
+      <div *ngIf="record.explanation">
+        <a class="badge badge-info collapsed"
+            data-toggle="collapse" href="#{{'score'+record.metadata.pid }}"
+            aria-expanded="false">
+            score: {{ record.explanation.value }}
+        </a>
+        <pre class="collapse border border-secondary mt-1" id="{{'score'+record.metadata.pid }}">{{record.explanation|json}}</pre>
```
Comment (Garfield-fr) [inline projects/public-search/src/app/document-brief/document-brief.component.html:66]: Missing space  after `{{` on first element and second missing space before ` }}` and after `{{`

File: projects/public-search/src/app/document-brief/document-brief.component.html [hunk 14]
```diff
@@ -16,62 +16,55 @@
...
-      </article>
-    </div>
+      <div *ngIf="record.metadata.type !== 'ebook'">
+        <i class="fa fa-circle text-{{ record.metadata.available ? 'success' : 'danger' }}" aria-hidden="true"></i>&nbsp;
+        <span translate *ngIf="record.metadata.available">available</span>
+        <span translate *ngIf="!record.metadata.available">not available</span>
+      </div>
+      <div *ngIf="record.explanation">
+        <a class="badge badge-info collapsed"
+            data-toggle="collapse" href="#{{'score'+record.metadata.pid }}"
```
Comment (Garfield-fr) [inline projects/public-search/src/app/document-brief/document-brief.component.html:62]: Missing space before ` }}` and after `{{`

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.spec.ts [hunk 15]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (Garfield-fr) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.spec.ts:1]: Missing License

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.scss [hunk 16]
```diff
@@ -0,0 +1,25 @@
+.thumb-brief
```
Comment (Garfield-fr) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.scss:1]: Missing License

**Review by iGormilhit — COMMENTED** [review]
Is it cover thumbnail? If so, the commit title could be amended.

**Review by iGormilhit — DISMISSED** [review]
~I think you're closing https://github.com/rero/rero-ils/issues/1396~

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html (hunk 1 above)
Comment (AoNoOokami) [reply projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html]: Done

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html (hunk 2 above)
Comment (AoNoOokami) [reply projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html]: Done

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts (hunk 4 above)
Comment (AoNoOokami) [reply projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:102]: Done

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts (hunk 5 above)
Comment (AoNoOokami) [reply projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:105]: Done

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html (hunk 9 above)
Comment (AoNoOokami) [reply projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:32]: I do not understand this comment. Can you explain again?

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.spec.ts (hunk 15 above)
Comment (AoNoOokami) [reply projects/shared/src/lib/view/thumbnail/thumbnail.component.spec.ts:1]: Done

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.scss (hunk 16 above)
Comment (AoNoOokami) [reply projects/shared/src/lib/view/thumbnail/thumbnail.component.scss:1]: Done

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.html (hunk 6 above)
Comment (AoNoOokami) [reply projects/shared/src/lib/view/thumbnail/thumbnail.component.html:26]: `transate`is for holidays, no? Ok I leave ->[]

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.html (hunk 7 above)
Comment (AoNoOokami) [reply projects/shared/src/lib/view/thumbnail/thumbnail.component.html:3]: Done

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts (hunk 8 above)
Comment (AoNoOokami) [reply projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:76]: Done

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

**Review by zannkukai — APPROVED** [review]

**Pull request merged**